moveit_core/collision_detection_bullet/test/test_bullet_continuous_collision_checking.cpp

```diff
@@ -0,0 +1,407 @@
+#include <gtest/gtest.h>
+#include <ros/ros.h>
+
+#include <moveit/collision_detection_bullet/bullet_integration/bullet_cast_bvh_manager.h>
+#include <moveit/collision_detection_bullet/bullet_integration/bullet_discrete_bvh_manager.h>
+#include <moveit/collision_detection/collision_common.h>
+
+#include <moveit/robot_model/robot_model.h>
+#include <moveit/robot_state/robot_state.h>
+#include <moveit/utils/robot_model_test_utils.h>
+
+#include <moveit/collision_detection_bullet/collision_world_bullet.h>
+#include <moveit/collision_detection_bullet/collision_robot_bullet.h>
+#include <moveit/collision_detection_bullet/bullet_integration/basic_types.h>
+
+#include <urdf_parser/urdf_parser.h>
+#include <geometric_shapes/shape_operations.h>
+
+/** \brief Brings the panda robot in user defined home position */
+inline void setToHome(robot_state::RobotState& panda_state)
+{
+  panda_state.setToDefaultValues();
+  double joint2 = -0.785;
+  double joint4 = -2.356;
+  double joint6 = 1.571;
+  double joint7 = 0.785;
+  panda_state.setJointPositions("panda_joint2", &joint2);
+  panda_state.setJointPositions("panda_joint4", &joint4);
+  panda_state.setJointPositions("panda_joint6", &joint6);
+  panda_state.setJointPositions("panda_joint7", &joint7);
+  panda_state.update();
+}
+
+class BulletCollisionDetectionTester : public testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    robot_model_ = moveit::core::loadTestingRobotModel("panda");
+    robot_model_ok_ = static_cast<bool>(robot_model_);
+
+    acm_.reset(new collision_detection::AllowedCollisionMatrix(robot_model_->getLinkModelNames(), false));
+
+    acm_->setEntry("panda_link0", "panda_link1", true);
+    acm_->setEntry("panda_link1", "panda_link2", true);
+    acm_->setEntry("panda_link2", "panda_link3", true);
+    acm_->setEntry("panda_link3", "panda_link4", true);
+    acm_->setEntry("panda_link4", "panda_link5", true);
+    acm_->setEntry("panda_link5", "panda_link6", true);
+    acm_->setEntry("panda_link6", "panda_link7", true);
+    acm_->setEntry("panda_link7", "panda_hand", true);
+    acm_->setEntry("panda_hand", "panda_rightfinger", true);
+    acm_->setEntry("panda_hand", "panda_leftfinger", true);
+    acm_->setEntry("panda_rightfinger", "panda_leftfinger", true);
+    acm_->setEntry("panda_link5", "panda_link7", true);
+    acm_->setEntry("panda_link6", "panda_hand", true);
```

Comment:
Can you update this to be the same code in "test_collision_common_panda.h"?

Reply:
yes, done.